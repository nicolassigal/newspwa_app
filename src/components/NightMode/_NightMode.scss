.night-mode-switch {
    position: absolute;
    right: 25px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    &__title {
        font-size: 14px;
        color: var(--txt-color);
    }
    input[type=checkbox]{
        height: 0;
        width: 0;
        visibility: hidden;
        & + label {
            &:after{
            background-image: url("../assets/sun.png");
            }
        }
        &:checked + label {
            background: $gold;
            &:after {
                left: calc(100% - 3px);
                transform: translateX(-100%);
                background-image: url("../assets/moon.png");
            }
        }
    }
    
    label {
        cursor: pointer;
        text-indent: -9999px;
        width: 50px;
        height: 30px;
        background: #e3e3e3;
        display: block;
        border-radius: 100px;
        position: relative;
        &:after {
            content: '';
            position: absolute;
            top: 3px;
            left: 3px;
            width: 25px;
                height: 25px;
            background-size: 100%;
            border-radius: 30px;
            transition: 0.3s;
        }
    }
}