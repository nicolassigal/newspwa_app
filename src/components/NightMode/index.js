import React, {createRef} from 'react';

const NightMode = () => {
    let mode = localStorage.getItem('theme');
    let el = createRef();
    document.documentElement.setAttribute('data-theme', mode);

    const switchMode = (evt) => {
       mode = el.checked ? 'dark': 'light';
       localStorage.setItem('theme', mode);
       document.documentElement.setAttribute('data-theme', mode);
    }
    
    return (
        <div className="night-mode-switch">
            <input 
            type="checkbox"
            ref={(input) => {el = input;}}
            onChange={switchMode}
            defaultChecked={mode === 'dark'}
            id="nightmode"/>
            <label htmlFor="nightmode">on</label>
         </div>
    )
}

export default NightMode;