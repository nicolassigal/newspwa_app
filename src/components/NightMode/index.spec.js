import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NightMode from './index';
import { shallow, mount } from 'enzyme';

describe('NightMode component test', () => {
  it('NightMode component should render correctly', () => {
   const wrapper = shallow(<NightMode />)
   expect(wrapper).toMatchSnapshot();
  });
});