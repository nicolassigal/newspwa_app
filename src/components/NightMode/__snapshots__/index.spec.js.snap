// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NightMode component test NightMode component should render correctly 1`] = `ShallowWrapper {}`;
